tests/vmi_configuration_test.go

```diff
@@ -2306,7 +2306,7 @@ var _ = Describe("Configurations", func() {
 			By("Check value in VM with dmidecode")
 			// Check on the VM, if expected values are there with dmidecode
 			res, err := expecter.ExpectBatch([]expect.Batcher{
-				&expect.BSnd{S: "[ $(sudo dmidecode -s chassis-asset-tag | tr -s ' ') -eq Test-123 ] && echo 'pass' || echo 'fail'\n"},
+				&expect.BSnd{S: "[ $(sudo dmidecode -s chassis-asset-tag | tr -s ' ') = Test-123 ] && echo 'pa''ss'\n"},
```

Comment:
why does it print `pa` and then `ss` ?

Reply:
What is typed into a terminal is instantly echoed back (so that a human user can see what they type). That causes BExp to "see" what was "typed" by BSnd.
Therefore, without my PR, `BExp{R: "pass"}` would always succeed, because it would see the word "pass" in `echo 'pass'`.
By splitting the word "pass" in 2, the result is unchanged, but the command doesn't contain the word "pass" anymore, so if BExp sees "pass" it really means that the test passed.

Reply:
i see, we saw this at #3543 with echo $? which expected 0 and 0 was part of the input of the ip so as you said
it saw what it sent,
the solution was to wait for \n0 instead of 0,
@AlonaKaplan is doing a general method for that , so you might want to sync with her

Reply:
I guess you mean this general approach: #3882

Reply:
Interesting, I missed that PR, I'll have a look, thanks.